### PR TITLE
SALTO-2577 use singular form for guide order types and instances, fix duplicate types

### PR DESCRIPTION
**Pull request by netama**
* Switch to `article_order`, `section_order`, `category_order` instead of the plural form, to conform with the existing order types and instances
* Avoid creating duplicate `x_order` types

---
_Release Notes_: 
None (not released yet)

---
_User Notifications_: 
None

### CODE REVIEW

**Review by idoser96 — COMMENTED** [review]
Also, why change from plural if these are lists of many elements?

**Review by idoser96 — APPROVED** [review]
Have a few questions but it looks good

File: packages/zendesk-adapter/src/filters/guide_order/article_order.ts [hunk 1]
```diff
@@ -47,10 +41,11 @@ const filterCreator: FilterCreator = ({ client, config }) => ({
       .filter(e => e.elemID.typeName === SECTION_TYPE_NAME)
 
     const orderType = createOrderType(ARTICLE_TYPE_NAME)
+    _.remove(elements, e => e.elemID.getFullName() === orderType.elemID.getFullName())
```
[the remark this answers is not in the document]
Comment (netama) [reply packages/zendesk-adapter/src/filters/guide_order/article_order.ts:44]: one of them is generated by the infra after a fix @edenhassid [added](https://github.com/salto-io/salto/pull/3529/files#diff-a1051ecabcee333024902369132c64d1396c2a57075d94370f1ffa153eb27cc1R1899) for that fixed an issue with how the elem ids were defined (see the change in the [adapter tests](https://github.com/salto-io/salto/pull/3548/files#diff-a8c8df720d5904cb467f2b4d39e52c63c6f3bfa8b0550b70312107fa72f472a5L164-L165) that removes the duplicates)

File: packages/zendesk-adapter/test/filters/guide_orders.test.ts [hunk 2]
```diff
@@ -132,7 +128,7 @@ const testFetch = async ({ createParent, createChild, orderField }
   const elements = [parentInstance, ...childInstances]
   await filter.onFetch(elements)
 
-  const typeObject = createOrderType(childInstances[0].elemID.typeName)
+  const typeObject = createOrderType(childInstances[0].elemID.typeName as 'article' | 'category' | 'section')
```
[the remark this answers is not in the document]
Comment (netama) [reply packages/zendesk-adapter/test/filters/guide_orders.test.ts:131]: oops, leftover from an attempt at adding safeties, removing

**Comment by netama** [comment]
re the question about plural - this is just not valid english :) the `order` part already says they are plural. and we already have `trigger_order` etc in the support part, we should follow the same pattern here

**Pull request merged**